modelcontextprotocol/server.py

```diff
@@ -908,6 +911,261 @@ def create_glossary_categories(categories) -> List[Dict[str, Any]]:
+                    - package_author, package_description, package_repository: Package metadata
+                    - workflow_steps: Workflow steps (only when getting by ID)
+                    - workflow_spec: Full workflow specification (only when getting by ID)
+            - total: Total count of workflows (1 when getting by ID, actual count when getting by type)
+            - error: None if no error occurred, otherwise the error message
+
+    Examples:
+        # Get a specific workflow by ID (full details with steps and spec)
+        result = get_workflows_tool(id="atlan-snowflake-miner-1714638976")
+        workflow = result.get("workflows", [])[0]  # Single workflow in list
+        # workflow contains complete workflow definition with workflow_steps and workflow_spec
+
+        # Get Snowflake workflows by type (metadata only)
+        result = get_workflows_tool(workflow_package_name="atlan-snowflake")
+        workflows = result.get("workflows", [])  # List of workflows
+
+        # Get BigQuery workflows with custom limit
+        result = get_workflows_tool(workflow_package_name="atlan-bigquery", max_results=50)
+
+        # Get workflows using enum name
+        result = get_workflows_tool(workflow_package_name="SNOWFLAKE")
+    """
+    return get_workflows(
+        id=id,
+        workflow_package_name=workflow_package_name,
+        max_results=max_results,
+    )
+
+
+@mcp.tool()
+def get_workflow_runs_tool(
+    workflow_name: str | None = None,
+    workflow_phase: str | None = None,
+    status: str
+    | None = None,  # This is technically a list but we get the input as a string so we parse it into a list
+    started_at: str | None = None,
+    finished_at: str | None = None,
+    from_: int = 0,
+    size: int = 100,
+) -> Dict[str, Any]:
+    """
+    Retrieve workflow_runs with flexible filtering options.
+
+    Naming Convention Note: The Argo/Kubernetes API uses terminology that can be confusing:
+    - "WorkflowTemplate" (kind="WorkflowTemplate") = workflow (template definition)
+    - "Workflow" (kind="Workflow") = workflow_run (executed instance/run)
+
+    This tool supports two main use cases:
+    1. Filter by specific workflow name and phase (single workflow)
+    2. Filter across all workflows by status list and optional time ranges
+
+    Each workflow_run contains execution details, timing, resource usage, and status information.
+
+    IMPORTANT: This tool returns only metadata and execution status. It does NOT return full workflow
+    instructions, steps, or transformations. To get complete workflow details including all steps and
+    transformations, use get_workflows_tool with an id parameter instead.
+
+    When to use this tool:
+    - Use when you need to find all workflow_runs of a specific workflow by execution phase
+        → Provide workflow_name AND workflow_phase (do NOT use status)
+    - Use when you need to find workflow_runs across multiple workflows filtered by status and time
```

Comment:
Can it get workflow runs across multiple workflows? Since it takes in workflow_name only